lib/src/client.dart

```diff
@@ -795,27 +795,49 @@ class Client extends MatrixApi {
         ));
       }
     }
+    try {
+      // Start a new direct chat
+      final roomId = await createRoom(
+        invite: [mxid],
+        isDirect: true,
+        preset: preset,
+        initialState: initialState,
+        powerLevelContentOverride: powerLevelContentOverride,
+      );
 
-    // Start a new direct chat
-    final roomId = await createRoom(
-      invite: [mxid],
-      isDirect: true,
-      preset: preset,
-      initialState: initialState,
-      powerLevelContentOverride: powerLevelContentOverride,
-    );
-
-    if (waitForSync) {
-      final room = getRoomById(roomId);
-      if (room == null || room.membership != Membership.join) {
-        // Wait for room actually appears in sync
-        await waitForRoomInSync(roomId, join: true);
+      if (waitForSync) {
+        final room = getRoomById(roomId);
+        if (room == null || room.membership != Membership.join) {
+          // Wait for room actually appears in sync
+          await waitForRoomInSync(roomId, join: true);
+        }
       }
-    }
 
-    await Room(id: roomId, client: this).addToDirectChat(mxid);
+      await Room(id: roomId, client: this).addToDirectChat(mxid);
 
-    return roomId;
+      return roomId;
+    } catch (e) {
```

Comment:
This is catching any error, not just the room creation error. You should narrowly scope the try catch and handle specific errors, not all of them and assume it is the one error you intended to handle.